nixos/modules/tasks/network-interfaces.nix

```diff
@@ -1417,8 +1446,10 @@ in
       serviceConfig.Type = "oneshot";
     };
 
-    environment.etc.hostid = mkIf (cfg.hostId != null) { source = hostidFile; };
-    boot.initrd.systemd.contents."/etc/hostid" = mkIf (cfg.hostId != null) { source = hostidFile; };
+    environment.etc.hostid = mkIf (cfg.hostId != null || cfg.machineId != null) { source = hostidFile; };
+    boot.initrd.systemd.contents."/etc/hostid" = mkIf (cfg.hostId != null || cfg.machineId != null) { source = hostidFile; };
```

Comment:
These should be able to just check `cfg.hostId`, since it's getting set based on `machineId` in its default.

Reply:
It should be resolved now, please check if I understood correctly :)